src/lib/utils/os_utils.cpp

```diff
@@ -614,6 +615,44 @@ void OS::page_named(void* page, size_t size) {
 #endif
 }
 
+#if defined(BOTAN_TARGET_OS_HAS_THREADS)
+void OS::set_thread_name(std::thread& thread, std::string& name) {
+   #if defined(BOTAN_TARGET_OS_IS_LINUX) || defined(BOTAN_TARGET_OS_IS_FREEBSD)
+   static_cast<void>(pthread_setname_np(thread.native_handle(), name.c_str()));
+   #elif defined(BOTAN_TARGET_OS_IS_OPENBSD)
+   static_cast<void>(pthread_set_name_np(thread.native_handle(), name.c_str()));
+   #elif defined(BOTAN_TARGET_OS_IS_NETBSD)
+   static_cast<void>(pthread_set_name_np(thread.native_handle(), "%s", const_cast<char*>(name.c_str())));
+   #elif defined(BOTAN_TARGET_OS_HAS_WIN32) && defined(BOTAN_BUILD_COMPILER_IS_MSVC)
+   const DWORD MS_VC_EXCEPTION = 0x406D1388;
```

Comment:
This approach doesn't seem so useful since it's works if you are currently attached to a debugger.

If it's possible to check at compile time what the target OS is, I'd suggest calling the Windows 10 function if it is available and for older systems just do nothing. If that's not possible I'd say just remove this block entirely and leave a comment that we can use `SetThreadDescription` in the future.

In 33dee2f4f I deprecated support for older Windows but we'll probably want to wait at least a release or two before pulling the trigger on it.